### PR TITLE
fix: Add huggingfaceml, swift, ansible and pub as valid packageTypes

### LINKED ISSUE #176: Include huggingfaceml as valid package type
Hi, would it be possible to include the "huggingfaceml" package type? 

`ArtifactoryRepository.get_repo(key)` returns a pydantic validation error with any huggingface remote repository:

```
  File "/home/user/.local/lib/python3.12/site-packages/pyartifactory/objects/repository.py", line 58, in get_repo
    return RemoteRepositoryResponse.model_validate(response_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/.local/lib/python3.12/site-packages/pydantic/main.py", line 551, in model_validate
    return cls.__pydantic_validator__.validate_python(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 1 validation error for RemoteRepositoryResponse
packageType
  Input should be 'alpine', 'bower', 'cargo', 'chef', 'cocoapods', 'composer', 'conan', 'conda', 'cran', 'debian', 'docker', 'gems', 'generic', 'gitlfs', 'go', 'gradle', 'helm', 'helmoci', 'ivy', 'maven', 'npm', 'nuget', 'opkg', 'p2', 'puppet', 'pypi', 'rpm', 'sbt', 'terraform', 'terraformbackend', 'vagrant', 'vcs' or 'yum' [type=enum, input_value='huggingfaceml', input_type=str]
    For further information 
```

I also noticed: swift, ansible and pub are missing.

### PR DESCRIPTION
**Pull request by Nexolight**
## Description

Updated PackageTypeEnum and added "huggingfaceml", "swift", "ansible" and "pub" as those are now supported in artifactory. 

Fixes #176

## Type of change

Please delete options that are not relevant.

- [Y] Bug fix (non-breaking change which fixes an issue)
- [Y] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has it been tested ?

Not well tested due to the nature of the change other than seeing if it would resolves the issue of calling `get_repository("repository_name")` on a repository with packageType "huggingfaceml" in conjunction with the hint in the official artifactory documentation that new ones will be added. 

Do not merge if there is any BaseRepositoryModel.packageType specific code that needs to be implemented.

## Checklist:

- [Y] My PR is ready for prime time! Otherwise use the ["Draft PR" feature](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests)
- [?] All commits have a correct title
- [n/a] Readme has been updated
- [Y] Quality tests are green (see Codacy).
- [Y] Automated tests are green (see pipeline)

### CODE REVIEW

File: pyartifactory/models/repository.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class PackageTypeEnum(str, Enum):
     """Enumerates package types."""
 
     alpine = "alpine"
+    ansible = "ansible"
```
Comment (anancarv) [inline pyartifactory/models/repository.py:16]: Ansible package type is missing from the [Artifactiory API ](https://jfrog.com/help/r/jfrog-rest-apis/repository-configuration-json). Can you please remove it?

Comment (Nexolight) [reply pyartifactory/models/repository.py:16]: I see it is not in there, but it is an option when creating a repository in the interface. 
Interestingly enough this works with the changes applied and the repository is created with the ansible package type as expected:

```
from pyartifactory import Artifactory
from pyartifactory.models.repository import RemoteRepository
art = Artifactory(
    url="http://docker-compose-artifactory.local:8082/artifactory", 
    auth=("*******","********"), 
    cert=None,
    api_version=2,
    verify=False
)
repo_created = art.repositories.create_repo(RemoteRepository(
    packageType="ansible",
    url="https://galaxy.ansible.com",
    key="testrepo-ansible"    
))
print(repo_created)
repo_req = art.repositories.get_repo("testrepo-ansible")
print(repo_req)

```
(Artifactory 7.90.5 rev 79005900)

Comment (anancarv) [reply pyartifactory/models/repository.py:16]: Yeah I see. I'm approving the PR, but can you open a support ticket for them to add ansible to their public documentation ?

**Review by anancarv — APPROVED** [review]

File: pyartifactory/models/repository.py (hunk 1 above)
Comment (Nexolight) [reply pyartifactory/models/repository.py:16]: Thank you, and done: [RTFACT-30747](https://jfrog.atlassian.net/browse/RTFACT-30747)

**Pull request merged**